apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/Session.cs

```diff
@@ -0,0 +1,175 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+
+namespace Google.Cloud.Spanner.V1
+{
+    public partial class Session : IDisposable
+    {
+        private readonly CancellationTokenSource _cancellation = new CancellationTokenSource();
+        private Task _keepAlive;
+        private int _lastAccessTime = Environment.TickCount;
```

Comment:
fixed overflow handling.  I think tickcount is easier because the timespan is fairly short and I would otherwise need to manage DST.